src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3414,6 +3414,37 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* node)
             blendVariableId = NI_EVEX_BlendVariableMask;
             op1             = maskNode;
         }
+        else if (op2->IsVectorZero() || op3->IsVectorZero())
+        {
+            // If either of the value operands is const zero, we can optimize down to AND or AND_NOT.
+            GenTree* binOp = nullptr;
+
+            if (op3->IsVectorZero())
+            {
+                binOp = comp->gtNewSimdBinOpNode(GT_AND, simdType, op1, op2, simdBaseJitType, simdSize);
+                BlockRange().Remove(op3);
+            }
+            else
+            {
+                binOp = comp->gtNewSimdBinOpNode(GT_AND_NOT, simdType, op3, op1, simdBaseJitType, simdSize);
```

Comment:
Is the ordering `op3, op1` significantly better or is the swapping of argument order vs. the `op1, op2` above "just because"?

Reply:
The second operand is the one that is NOT'ed. It has to the mask, which is `op1`.

Reply:
Cool, thanks! This is all new to me 😔